From 783dd118ac8d2f83233da4c3a0c84c35b2b0e885 Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Fri, 19 Mar 2021 11:24:25 +0200
Subject: [PATCH 08/42] fsl_esdhc_imx: Fix compile error when
 !MMC_SUPPORTS_TUNING

usdhc_s32gen1_data should not depend on MMC_SUPPORTS_TUNING.

Upstream-Status: Pending

Issue: ALB-6672
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/mmc/fsl_esdhc_imx.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/fsl_esdhc_imx.c b/drivers/mmc/fsl_esdhc_imx.c
index fbdefd71af..7779e0d22a 100644
--- a/drivers/mmc/fsl_esdhc_imx.c
+++ b/drivers/mmc/fsl_esdhc_imx.c
@@ -686,6 +686,8 @@ static void set_sysctl(struct fsl_esdhc_priv *priv, struct mmc *mmc, uint clock)
 	priv->clock = clock;
 }
 
+static struct esdhc_soc_data usdhc_s32gen1_data;
+
 #ifdef MMC_SUPPORTS_TUNING
 static int esdhc_change_pinstate(struct udevice *dev)
 {
@@ -861,7 +863,6 @@ static int esdhc_set_voltage(struct mmc *mmc)
 	}
 }
 
-#ifdef MMC_SUPPORTS_TUNING
 static int fsl_s32gen1_esdhc_execute_tuning(struct udevice *dev,
 					    uint32_t opcode)
 {
@@ -941,8 +942,6 @@ static void esdhc_stop_tuning(struct mmc *mmc)
 	dm_mmc_send_cmd(mmc->dev, &cmd, NULL);
 }
 
-static struct esdhc_soc_data usdhc_s32gen1_data;
-
 static int fsl_esdhc_execute_tuning(struct udevice *dev, uint32_t opcode)
 {
 	struct fsl_esdhc_plat *plat = dev_get_platdata(dev);
@@ -1042,7 +1041,6 @@ static int fsl_esdhc_execute_tuning(struct udevice *dev, uint32_t opcode)
 
 	return ret;
 }
-#endif
 #endif /* MMC_SUPPORTS_TUNING */
 
 static int esdhc_set_ios_common(struct fsl_esdhc_priv *priv, struct mmc *mmc)
-- 
2.25.1

